datasource/src/main/scala/quasar/physical/mongo/Mongo.scala

```diff
@@ -99,19 +101,22 @@ object Mongo {
     }
 
     def enqueueObservable(
-        obsQ: NoneTerminatedQueue[F, Either[Throwable, F[A]]],
+        obsQ: NoneTerminatedQueue[F, Either[Throwable, A]],
         subVar: MVar[F, Subscription])
         : Stream[F, Unit] =
       Stream.eval(F.delay(handler(subVar, { x => run(obsQ.enqueue1(x)) })))
 
     (for {
-      obsQ <- Stream.eval(Queue.boundedNoneTerminated[F, Either[Throwable, F[A]]](1))
+      obsQ <- Stream.eval(Queue.boundedNoneTerminated[F, Either[Throwable, A]](queueSize))
       subVar <- Stream.eval(MVar[F].empty[Subscription])
       _ <- enqueueObservable(obsQ, subVar)
-      res <- obsQ.dequeue.rethrow.onFinalize(subVar.take.flatMap(unsubscribe))
-    } yield Stream.eval(res)).flatten
+      res <- obsQ.dequeue.chunkN(queueSize, true).flatMap( c =>
```

Comment:
If we want to emit sized chunks, instead of using `chunkN`, I think it would be more efficient to accumulate into a sized `ArrayBuffer` in `onNext` and enqueue into a `Queue[Either[Throwable, Chunk[A]]]` once the buffer is full. That way we don't have to worry about resizing/splitting.

Reply:
dequeueChunks is more efficient still.

Reply:
I've tried both. `ArrayBuffer` works a bit slower, `dequeuChunks.chunks` works with the same speed as `dequeue.chunkN`

Reply:
The fact that `dequeueChunks.chunks` performs the same as `dequeue.chunkN` is a coincidence caused by the speed of your local machine. On real latencies, it will out-perform `chunkN` due to the fact that its chunk boundaries are variable based on availability, rather than fixed.